packages/components/src/confirm-dialog/test/index.js

```diff
@@ -140,6 +154,7 @@ describe( 'Confirm', () => {
 				const confirmDialog = screen.getByRole( 'dialog' );
 
 				//The overlay click is handled by detecting an onBlur from the modal frame.
+				// TODO: replace with `@testing-library/user-event`
```

Comment:
I tried to simulate a click on `document.body` and even on the Modal's overlay, but that was not enough. I therefore decided to leave `fireEvent` for now to unblock the rest of this PR (same for another test a few lines below)